Create and save student with null English Proficiency

nondescryptid opened the pull request:
Fixes #84 by deleting EnglishProficiency from the request body to POST/PATCH whenever EnglishProficiency is "" or null. 

We couldn't post students with no EngProf from the frontend as an empty EngProf would be sent to the backend as EnglishProficiency: "" instead of not being sent at all. Then it'd fail as it violates the constraints for EngProf (as "" is not one of the enum values).

nondescryptid committed:
edited students.js to delete EnglishProfieicncy from request body if EngProf is null or empty

nondescryptid committed:
Merging master to avoid conflicts
Merge branch 'master' into studentNullEngProf

nondescryptid committed:
Merging master to avoid conflicts
Merge branch 'master' into studentNullEngProf

therealyingtong commented on backend/src/routes/students.js:
```diff
@@ -61,6 +61,10 @@ router.get('/:id', async (req, res) => {
 /* POST students listing */
 router.post('/', async (req, res) => {
+  // If EnglishProficiency is empty
+  if (req.body.EnglishProficiency === "" || null) {
```
Due to [logical operator precedence](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence), `if (string === "something" || null)` is parsed as `(if string === "something") OR (if null)`, whereas we want `if string === ("something" OR null)`

```suggestion
  if (!req.body.EnglishProficiency || req.body.EnglishProficiency === "") {
```

lauratsui reviewed (commented):
LGTM pending yingtong's comment, thanks!

nondescryptid committed:
change logical operator to check if req.body.EnglishProficiency is null or empty

Co-authored-by: ying tong <yingtong@z.cash>

nondescryptid committed:
Merge master into branch to avoid merge conflicts
Merge branch 'master' into studentNullEngProf

nondescryptid commented:
Thank you @therealyingtong for catching the issue with logical operators and laura for reviewing! i've committed the changes such that it checks that the contents of req.body.EnglishProficiency are null

nondescryptid committed:
committed change to logical operators checking if req.body.EnglishProficiency is null
Merge branch 'studentNullEngProf' of github.com:szktm/irrc-madwish into studentNullEngProf
